node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -45,30 +44,117 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
+  def peerData: Map[Peer, PeerData] = _peerData.toMap
+
+  def peers: Vector[Peer] = peerData.keys.toVector
+
+  /** Returns peers by querying each dns seed once. These will be IPv4 addresses. */
+  def getPeersFromDnsSeeds: Vector[Peer] = {
+    val dnsSeeds = nodeAppConfig.network.dnsSeeds
+    val addresses = dnsSeeds
+      .flatMap(seed => {
+        try {
+          InetAddress
+            .getAllByName(seed)
+        } catch {
+          case _: UnknownHostException =>
+            logger.debug(s"DNS seed $seed is unavailable")
+            Vector()
+        }
+      })
+      .distinct
+      .filter(_.isReachable(500))
+      .map(_.getHostAddress)
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers.toVector
+  }
 
-  def setPeerServices(
-      peer: Peer,
-      serviceIdentifier: ServiceIdentifier): Unit = {
-    _peerServices.put(peer, serviceIdentifier)
-    ()
+  /** Returns peers from hardcoded addresses taken from https://github.com/bitcoin/bitcoin/blob/master/contrib/seeds/nodes_main.txt */
+  def getPeersFromResources: Vector[Peer] = {
+    val source = Source.fromURL(getClass.getResource("/hardcoded-peers.txt"))
+    val addresses = source
+      .getLines()
+      .toVector
+      .filter(nodeAppConfig.torConf.enabled || !_.contains(".onion"))
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  /** Returns all peers stored in db */
+  def getPeersFromDb: Future[Vector[Peer]] = {
+    val addressesF: Future[Vector[PeerDb]] =
+      PeerDAO().findAll()
+    val peersF = addressesF.map { addresses =>
+      val filteredAddresses = addresses.filter(
+        nodeAppConfig.torConf.enabled || _.networkId != AddrV2Message.TOR_V3_NETWORK_BYTE)
+      val inetSockets = filteredAddresses.map(a => {
+        NetworkUtil.parseInetSocketAddress(a.address, a.port)
+      })
+      val peers =
+        inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+      peers
+    }
+    peersF
+  }
+
+  def getPeersFromConfig: Vector[Peer] = {
+    val addresses = nodeAppConfig.peers.filter(
+      nodeAppConfig.torConf.enabled || !_.contains(".onion"))
+    val inetSockets = addresses.map(
+      NetworkUtil.parseInetSocketAddress(_, nodeAppConfig.network.port))
+    val peers =
+      inetSockets.map(Peer.fromSocket(_, nodeAppConfig.socks5ProxyParams))
+    peers
+  }
+
+  /** Returns peers randomly taken from config, db, hardcoded peers, dns seeds in that order */
+  def getPeers: Future[Vector[Peer]] = {
+    //currently this would only give the first peer from config
+    val peersFromConfig = getPeersFromConfig
+    lazy val peersFromDbF = getPeersFromDb
```

Comment:
why are these `lazy` ?

Reply:
lazy on `peersFromDbF` isn't really needed since we resolve that future anyway, I will remove that. For the rest, lazy is to avoid calling those functions unless needed, particularly on `getPeersFromDnsSeeds` as that takes a while to complete.